### PR TITLE
Miscellaneous cleanup/throw helpers

### PR DESCRIPTION
**Pull request by Rob-Hague**
These are some random changes I've accumulated in some branches, which I've extracted out to reduce diffs & conflicts. They are roughly:

Add a reference to PolySharp to allow nullable attributes and CallerArgumentExpressionAttribute etc.
Add a ThrowHelpers class
Add #nullable to PrivateKeyFile and cleanup PrivateKeyFileTest
Also change around some of the newly added keys - I have a use for this with certificate support.

### CODE REVIEW

**Comment by Rob-Hague** [comment]
Ok to get a sign-off here?

File: Directory.Build.props [hunk 1]
```diff
@@ -32,7 +32,7 @@
         Disable nullable warnings on old frameworks because of missing annotations.
   -->
   <PropertyGroup Condition=" !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0')) ">
-    <NoWarn>$(NoWarn);CS8602</NoWarn>
+    <NoWarn>$(NoWarn);CS8602;CS8604;CS8777</NoWarn>
```
Comment (drieseng) [inline Directory.Build.props:35]: Not possible to fix these?

File: src/Renci.SshNet/.editorconfig [hunk 2]
```diff
@@ -188,3 +188,6 @@ dotnet_diagnostic.MA0040.severity = none
 # MA0042: Do not use blocking calls in an async method
 # duplicate of CA1849
 dotnet_diagnostic.MA0042.severity = none
+
+# S3236: Caller information arguments should not be provided explicitly
+dotnet_diagnostic.S3236.severity = none
```
Comment (drieseng) [inline src/Renci.SshNet/.editorconfig:193]: Why was this one reported? Don't have time to check myself now.

**Review by drieseng — APPROVED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (Rob-Hague) [reply Directory.Build.props:35]: They are coming from string.IsNullOrEmpty and friends which are not annotated for nullable analysis on lower targets

File: src/Renci.SshNet/.editorconfig (hunk 2 above)
Comment (Rob-Hague) [reply src/Renci.SshNet/.editorconfig:193]: It comes when one wants to provide a value for a parameter marked with `[CallerExpressionAttribute]`, e.g.

```c#
public TimeSpan Timeout
{
    get
    {
        return _timeout;
    }
    set
    {
        // EnsureValidTimeout(this TimeSpan timeSpan, [CallerArgumentExpression(nameof(timeSpan))] string? paramName = null)

        value.EnsureValidTimeout(nameof(Timeout));

        _timeout = value;
    }
}
```

I would probably suppress this one at the call sites, but #1480 will also bring more places where we don't want it (from Debug.Assert)

**Comment by Rob-Hague** [comment]
Thanks

**Comment by scott-xu** [comment]
> Also change around some of the newly added keys - I have a use for this with certificate support.

How about consolidating the keys. e.g. one key-pair per algorithm (and key size) , but with different format and different encryption for private key? It will be easier for integration test.

**Comment by Rob-Hague** [comment]
It seems OK how it is - it was a bit of a pain to set everything up for #1498 so I would rather like to avoid changing it

**Pull request merged**